signingscript/src/signingscript/sign.py

```diff
@@ -558,16 +557,26 @@ def _get_omnija_signing_files(file_list):
     return files
 
 
-# _run_generate_precomplete {{{1
-def _run_generate_precomplete(context, tmp_dir):
+# _update_precomplete {{{1
+def _update_precomplete(context, tmp_dir):
     """Regenerate `precomplete` file with widevine sig paths for complete mar."""
     log.info("Generating `precomplete` file...")
-    path = _ensure_one_precomplete(tmp_dir, "before")
-    with open(path, "r") as fh:
+    precomplete = _ensure_one_precomplete(tmp_dir)
+    with open(precomplete, "r") as fh:
         before = fh.readlines()
-    generate_precomplete(os.path.dirname(path))
-    path = _ensure_one_precomplete(tmp_dir, "after")
-    with open(path, "r") as fh:
+    with open(precomplete, "w") as fh:
+        for line in before:
+            fh.write(line)
+            instr, path = line.strip().split(None, 1)
+            if instr != "remove":
+                continue
+            if not path or path[0] != '"' or path[-1] != '"':
+                continue
+            file = path[1:-1]
+            if _get_widevine_signing_files([file]):
+                sigfile = _get_mac_sigpath(file)
+                fh.write('remove "{}"\n'.format(sigfile))
```

Comment:
I'm not worrying that this could add a duplicate remove instruction if the file was already signed in the input archive.